advance_check_void/models/account_payment.py

```diff
@@ -0,0 +1,391 @@
+        check_hist_obj = self.env["payment.check.history"]
+        journal_id = self.journal_id and self.journal_id.id
+        self._cr.execute(
+            """
+                SELECT DISTINCT(aml.move_id)
+                FROM account_payment AS ap
+                INNER JOIN account_move_line AS aml
+                    ON aml.payment_id=ap.id
+                    AND ap.id = %s
+                INNER JOIN account_move AS am
+                    ON aml.move_id = am.id
+            """,
+            (self.id,),
+        )
+        am_ids = self._cr.fetchone()
+        if am_ids:
+            am_ids = am_ids[0]
+
+        for rec in self:
+            # Update payment check history..
+            check_ids = check_hist_obj.search(
+                [("payment_id", "=", rec.id), ("check_number", "=", rec.check_number)],
+                order="id desc",
+                limit=1,
+            )
+            if check_ids:
+                check_ids.write({"state": "posted"})
+            self.write({"is_move_sent": False})
+        return {
+            "name": _("Reverse Moves"),
+            "type": "ir.actions.act_window",
+            "res_model": "account.move.reversal",
+            "view_type": "form",
+            "view_mode": "form",
+            "target": "new",
+            "context": {
+                "is_void_button": True,
+                "default_journal_id": journal_id or False,
+                "account_move_id": am_ids,
+                "payment_id": self.id,
+            },
+        }
+
+    def button_void_checks(self):
+        return {
+            "name": _("Void Checks"),
+            "view_type": "form",
+            "view_mode": "tree,form",
+            "res_model": "payment.check.void",
+            "view_id": False,
+            "type": "ir.actions.act_window",
+            "domain": [("payment_id", "in", self.ids)],
+        }
+
+    def action_validate_invoice_payment(self):
+        for record in self:
+            if record.payment_type == "inbound" and record.payment_method_code in (
+                "check_printing",
+                "ACH-Out",
+            ):
+                if record.check_number <= 0:
+                    raise ValidationError(_("Please Enter Check Number"))
+        return super(AccountPayment, self).action_validate_invoice_payment()
+
+    def post(self):
+        res = super(AccountPayment, self).post()
+        for record in self:
+            if (
+                record.payment_type == "inbound"
+                and record.payment_method_code
+                in ("check_printing", "ACH-Out", "manual")
+                and not record.invoice_ids
+                and record.check_number <= 0
+            ):
+                raise ValidationError(_("Please Enter Check Number"))
+        return res
+
+
+class PaymentCheckHistory(models.Model):
+    _name = "payment.check.history"
+    _description = "Payment Info for the check Payment Feature"
+    _order = "write_date desc"
```

Comment:
You set `write_date desc` default order, but then force `id desc` on every search. Why is that?
Should it be `id desc` in the first place?